homeassistant/components/esphome/manager.py

```diff
@@ -538,7 +538,7 @@ async def async_start(self) -> None:
             on_connect=self.on_connect,
             on_disconnect=self.on_disconnect,
             zeroconf_instance=self.zeroconf_instance,
-            name=self.host,
+            name=entry.data.get(CONF_DEVICE_NAME) or self.host,
```

Comment:
```suggestion
            name=entry.data.get(CONF_DEVICE_NAME, self.host),
```

Reply:
I was a bit concerned that `CONF_DEVICE_NAME` could be `""`

Reply:
looks like esphome rejects it so we can change it
`InvalidEnvNameError: Invalid environment name ''. The name can contain alphanumeric, underscore, and hyphen characters (a-z, 0-9, -, _)`